src/main/scala/exercises/04_Options.scala

```diff
@@ -33,17 +33,26 @@ object Options {
    * - empty:  return "empty"	        E.g. None is "empty"
    * - does not exist: 					"not existing"
    */
-  def roomState(rooms: Map[Int, Option[String]], room: Int): String = {
-    error("Fix me")
-  }
+  def roomState(rooms: Map[Int, Option[String]], room: Int): String = 
+    rooms.get(room) match {
+      case None                 ⇒ "not existing"
+      case Some(None)           ⇒ "empty"
+      case Some(Some("locked")) ⇒ "not available"
+      case Some(Some(s))        ⇒ s
+    }
 
```

Comment:
This is the only class where I've seen non-ASCII symbols.  I worry that certain computers won't be able to understand these characters (I remember there being a problem on Knuth with certain settings), but it also makes the code a bit nicer to read.